README.md

```diff
@@ -0,0 +1,79 @@
+# Spring reCaptcha
+
+Library providing a custom filter to validate reCaptcha
+
+## Requirements
+
+* JDK 11 +
+* Spring Boot MVC application
+
+## Supported reCaptcha version
+
+* [X] reCaptcha v2
+* [ ] reCaptcha v3
+
+## Usage 
+
+To add the custom filter, you need to include the dependency and configure required property (see [Required configuration](#required-configuration)).
+
+```kotlin
+dependencies {
+  ...
+  implementation("com.ekino.oss.spring:spring-recaptcha:0.0.1-SNAPSHOT")
+  ...
+}
+```
+
+## Configuration
+
+### Required configuration
+
+The library comes with a configuration by default.
+However, you must configure the secret before using it.
+
+```yaml
+security:
+  recaptcha:
+    secret: <your_secret>
+```
+
+### Other configuration
+
+#### Disable filter
+
+ReCatpcha validation filter can be disabled by setting the property :
+
+```yaml
+security:
+  recaptcha:
+    enabled: false
+``` 
+
+#### Filter methods and urls
+
+By default, every url for the `POST` method will need reCaptcha validation, you can filter urls and methods by setting the properties : 
+```yaml
+security:
+  recaptcha:
+    url-patterns:
+      - "/resources/**
+    filtered-method:
+      - POST
+      - PUT
+```  
+
+#### By-passing the filter
+
+Connecting or validation issues could occurs with external services. 
+To prevent re-deployment of the whole application to disabled the filter, you can configure a private key and send it in 
+a custom header `X-ReCaptcha-ByPass-Key` to by-pass the reCaptcha validation.
+
+```yaml
+security:
+  recaptcha:
+    by-pass-key: <your_by_pass_private_key>  
```

Comment:
Where does this idea come from ?

Reply:
From Sergio to have a quick way to byPass the filter in case of error in production

Reply:
ok, fine !